internal/oci/container.go

```diff
@@ -160,7 +160,14 @@ func (c *Container) FromDisk() error {
 	defer jsonSource.Close()
 
 	dec := json.NewDecoder(jsonSource)
-	return dec.Decode(c.state)
+	var state ContainerState
+	if err := dec.Decode(&state); err != nil {
+		return err
+	}
+	c.opLock.Lock()
+	c.state = &state
+	c.opLock.Unlock()
+	return nil
```

Comment:
IDK dec.Decode() and couldn't find it quickly to verify, but I think the value in state can change in that right?  If so, should the c.opLock.Lock() be above that call?

Reply:
state is local variable.
It is expected that its value changes after the decoding.

Reply:
DOH, that's right, the trick is it's a local variable and safe.  Sorry for the churn, thx @tedyu